qiskit/primitives/backend_sampler_v2.py

```diff
@@ -0,0 +1,199 @@
+# that they have been altered from the originals.
+
+"""Sampler V2 implementation for an arbitrary BackendV2 object."""
+
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass
+from typing import Iterable
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.primitives.backend_estimator import _run_circuits
+from qiskit.primitives.base import BaseSamplerV2
+from qiskit.primitives.containers import (
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPubLike,
+    make_data_bin,
+)
+from qiskit.primitives.containers.bit_array import _min_num_bytes
+from qiskit.primitives.containers.sampler_pub import SamplerPub
+from qiskit.primitives.primitive_job import PrimitiveJob
+from qiskit.providers.backend import BackendV2
+from qiskit.result import Result
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    start: int
+
+
+class BackendSamplerV2(BaseSamplerV2):
+    """
+    Implementation of :class:`BaseSamplerV2` using a backend.
+
+    This class provides a SamplerV2 interface from any :class:`~.BackendV2` backend
+    and doesn't do any measurement mitigation, it just computes the bitstrings.
+
+    This sampler supports providing arrays of parameter value sets to
+    bind against a single circuit.
+
+    Each tuple of ``(circuit, <optional> parameter values, <optional> shots)``, called a sampler
+    primitive unified bloc (PUB), produces its own array-valued result. The :meth:`~run` method can
+    be given many pubs at once.
+
+    .. note::
+
+        This class requires a backend that supports ``memory`` option.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        backend: BackendV2,
+        default_shots: int = 1024,
+    ):
+        """
+        Args:
+            backend: Required: the backend to run the sampler primitive on
+            default_shots: The default shots for the sampler if not specified during run.
+        """
+        super().__init__()
+        self._backend = backend
+        self._default_shots = default_shots
+
+    @property
+    def backend(self) -> BackendV2:
+        """Returns the backend which this sampler object based on."""
+        return self._backend
+
+    @property
+    def default_shots(self) -> int:
+        """Return the default shots"""
+        return self._default_shots
+
+    def run(
+        self, pubs: Iterable[SamplerPubLike], *, shots: int | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        if shots is None:
+            shots = self._default_shots
+        coerced_pubs = [SamplerPub.coerce(pub, shots) for pub in pubs]
+        self._validate_pubs(coerced_pubs)
+        job = PrimitiveJob(self._run, coerced_pubs)
+        job._submit()
+        return job
+
+    def _validate_pubs(self, pubs: list[SamplerPub]):
+        for i, pub in enumerate(pubs):
+            if len(pub.circuit.cregs) == 0:
```

Comment:
Do we need to check for measure instruction? Should we rather check measure instruction?

Reply:
I'm not sure whether we need more validations. This is basically same as the check of `StatevectorSampler`.
https://github.com/Qiskit/qiskit/blob/ff2bbfede9c64f6de63ffe9052c8fea7a4bc7e8d/qiskit/primitives/statevector_sampler.py#L161-L166